File: solutions/scala/gigasecond/2/src/main/scala/Gigasecond.scala
import java.time.{LocalDate, LocalDateTime}

object Gigasecond {
  val secondsPerGigasecond = 1000000000

  def add(startDate: LocalDate): LocalDateTime =
    this.add(startDate.atTime(0, 0))

  def add(startDateTime: LocalDateTime): LocalDateTime =
    startDateTime.plusSeconds(secondsPerGigasecond)
}

File: solutions/java/gigasecond/2/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public record Gigasecond(LocalDateTime dateTime) {
	private static final long GIGASECOND = 1_000_000_000L;

	public Gigasecond(LocalDateTime dateTime) {
		this.dateTime = dateTime.plusSeconds(GIGASECOND);
	}

	public Gigasecond(LocalDate moment) {
		this(moment.atStartOfDay());
	}

	@SuppressWarnings("unused")
	public LocalDateTime getDateTime() {
		return dateTime;
	}
}
